### PR TITLE
Feature/GitHub actions pipeline

### PR DESCRIPTION
**Pull request by DavidDamke**
Environment variable SONAR_TOKEN muss noch in doubleSlashde/KeepTime Github hinzugefügt werden.

### CODE REVIEW

File: .github/workflows/mavenCi.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+          distribution: 'corretto'
+          cache: maven
+      - name: Analyze
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}  #Added new Repository Secret in GitHub
+        run: mvn sonar:sonar
+          -Dsonar.host.url="https://sonarcloud.io"
+          -Dsonar.organization=daviddamke
+          -Dsonar.projectKey=DavidDamke_KeepTime -X
```
Comment (Death111) [inline .github/workflows/mavenCi.yml:50]: can you make org+projectkey also a pipeline variable. So we can adapt them easily (do not want to have someones account hardcoded)

File: .github/workflows/mavenCi.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      - uses: actions/checkout@v3
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v3
+        with:
+          java-version: '8'
+          distribution: 'corretto'
+          cache: maven
+      - name: Build
+        run: |
+          mvn package -B clean verify --file pom.xml
```
Comment (Death111) [inline .github/workflows/mavenCi.yml:28]: I guess we do not need to specify the pom file - as its the default one?

File: .github/workflows/mavenCi.yml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+          -Dsonar.host.url="https://sonarcloud.io"
+          -Dsonar.organization=daviddamke
+          -Dsonar.projectKey=DavidDamke_KeepTime -X
+          -Dsonar.java.binaries=.
+          -Dsonar.qualitygate.wait=true
+      - name: Upload Build Artifact
+        uses: actions/upload-artifact@v3
+        with:
+          # Artifact name
+          name: keeptime-1.2.0-SNAPSHOT-bin.zip
```
Comment (Death111) [inline .github/workflows/mavenCi.yml:57]: the artifact does not contain the expected content (.jar + licenses). instead it has a folder with sonar files?

**Review by Death111 — CHANGES_REQUESTED** [review]

File: .github/workflows/mavenCi.yml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+on:
+  workflow_dispatch:
+  push:
+  pull_request:
+
+
+
+jobs:
+  job1:
```
[the remark this answers is not in the document]
Comment (DavidDamke) [reply .github/workflows/mavenCi.yml]: Renamed the jobs  build and analyze

File: .github/workflows/mavenCi.yml (hunk 1 above)
Comment (DavidDamke) [reply .github/workflows/mavenCi.yml]: org+projectkey+url are now pipeline variables

File: pom.xml [hunk 5]
```diff
@@ -92,9 +92,38 @@
...
+      <groupId>org.sonarsource.scanner.maven</groupId>
+      <artifactId>sonar-maven-plugin</artifactId>
+      <version>3.9.1.2184</version>
+    </dependency>
   </dependencies>
   <build>
     <plugins>
+      <plugin>
+        <groupId>org.jacoco</groupId>
+        <artifactId>jacoco-maven-plugin</artifactId>
```
[the remark this answers is not in the document]
Comment (DavidDamke) [reply pom.xml:105]: Deleted the plugin

File: .github/workflows/mavenCi.yml (hunk 3 above)
Comment (DavidDamke) [reply .github/workflows/mavenCi.yml]: The artifact now contains the expected content

File: .github/workflows/mavenCi.yml (hunk 2 above)
Comment (DavidDamke) [reply .github/workflows/mavenCi.yml]: --file pom.xml is not necessary and is removed

File: src/main/java/de/doubleslash/keeptime/common/SvgNodeProvider.java [hunk 6]
```diff
@@ -20,10 +20,17 @@
 
 public class SvgNodeProvider {
 
-   public static String getSvgPathWithXMl(Resources.RESOURCE resource) {
+   public static String getSvgPathWithXMl(Resources.RESOURCE resource){
       String svgPath;
       Document document;
       DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
+      try {
+         dbf.setFeature("http://apache.org/xml/features/disallow-doctype-decl", true);
```
[the remark this answers is not in the document]
Comment (DavidDamke) [reply src/main/java/de/doubleslash/keeptime/common/SvgNodeProvider.java:28]: added Issue description + the Solution I used

File: .github/workflows/mavenCi.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (DavidDamke) [reply .github/workflows/mavenCi.yml]: changed the name to "KeepTime"

**Review by Death111 — CHANGES_REQUESTED** [review]

**Review by Death111 — APPROVED** [review]

**Pull request merged**